pkg/logql/log/labels.go

```diff
@@ -596,9 +607,12 @@ Outer:
 				continue Outer
 			}
 		}
-		for _, la := range b.add {
+		for category, la := range b.add {
 			for _, l := range la {
 				if g == l.Name {
+					if LabelCategory(category) == StructuredMetadataLabel {
+						b.referencedStructuredMetadata = true
+					}
```

Comment:
I believe these lines will set `b.referencedStructuredMetadata` to true whenever we have structured metadata extracted from the storage... even if the query does not use it. is it expected?

Reply:
This is for queries referencing structured metadata in `by` clause for example if you have `foo` in structured metadata then the flag would be turned on for query `sum by (foo) (...)`